src/CombatData.ts

```diff
@@ -245,6 +246,23 @@ export class CombatData {
     }
 
     switch (event.logLine.event) {
+      case LogEvent.SPELL_ABSORBED:
+        {
+          const absorbAction = event as CombatAbsorbAction;
+          // There is an edge case where the first spell of a match is a SPELL_ABSORBED
+          // event and the unit that cast the shield isn't registered in the units array yet
+          // In this case - add the unit to the list before attempting to push the event
+          if (!this.units[absorbAction.shieldOwnerUnitId]) {
+            this.units[absorbAction.shieldOwnerUnitId] = new CombatUnit(
+              absorbAction.shieldOwnerUnitId,
+              absorbAction.shieldOwnerUnitName
+            );
+          }
+          const shieldOwner = this.units[absorbAction.shieldOwnerUnitId];
+          shieldOwner.absorbsOut.push(absorbAction);
+          destUnit.absorbsIn.push(absorbAction);
```

Comment:
nitpick - "in" vs "out" can be a bit confusing here. does "out" mean I absorbed something or I casted something out that got absorbed?